dashboard/app/models/user.rb

```diff
@@ -1252,6 +1240,26 @@ def can_edit_password?
     encrypted_password.present?
   end
 
+  def can_edit_display_name?
+    !picture_or_word_account?
+  end
+
+  def picture_or_word_account?
```

Comment:
As discussed, I'll maintain that your account type is dictated by `provider`, nothing else.  This logic captures (intends to capture) some other concept, perhaps `user_owned_account?`.

That said, I don't think this logic captures the intent. For example, this logic would suggest that an orphaned picture account (e.g., the teacher deletes the section) is *not* a picture or word account, whereas (at least in my mind), the method should be returning `true` in this case.

Note that `sections_as_student` scopes to non-deleted sections and followers. Is this desired?